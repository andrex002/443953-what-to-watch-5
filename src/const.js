const filmsCount = {
  ALL: 30,
  PER_STEP: 8,
  SIMILAR: 4
};

const TITLES = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`,
  `Shutter Island`,
  `Pulp Fiction`,
  `Snatch`,
  `War of the Worlds`,
  `Mindhunter`,
  `Dardjeeling Limited`
];

const POSTERS = [
  `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  `bohemian-rhapsody.jpg`,
  `macbeth.jpg`,
  `aviator.jpg`,
  `we-need-to-talk-about-kevin.jpg`,
  `what-we-do-in-the-shadows.jpg`,
  `revenant.jpg`,
  `johnny-english.jpg`,
  `shutter-island.jpg`,
  `pulp-fiction.jpg`,
  `snatch.jpg`,
  `war-of-the-worlds.jpg`,
  `mindhunter.jpg`,
  `dardjeeling-limited.jpg`
];

const GENRES = [`Crime`, `Comedy`, `Drama`, `Documentary`, `Horror`, `Kids & Family`, `Romance`, `Sci-Fi`, `Thriller`];

const DIRECTORS = [`James Cameron`, `Alfred Hitchcock`, `Steven Spielberg`];

const ACTORS = [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`, `Tony Revoloru`, `Bruce Lee`, `VanDamme`, `Jacke Chan`, `Tilda Swinton`, `Tom Wilkinson`, `Owen Wilkinson`, `Adrien Brody`, `Ralph Fiennes`, `Jeff Goldblum`];

const DESCRIPTION_TEXTS = [
  `It’s only after we’ve lost everything that we’re free to do anything.`,
  `It is not our abilities that show what we truly are… it is our choices.`,
  `Oh yes, the past can hurt. But you can either run from it, or learn from it.`,
  `Do, or do not. There is no “try”`,
  `Life moves pretty fast. If you don’t stop and look around once in a while, you could miss it.`,
  `I believe whatever doesn’t kill you, simply makes you…stranger.`,
  `The truth is a matter of circumstances, it’s not all things to all people all the time.`,
  `When I despair, I remember that all through history the way of truth and love has always won. There have been tyrants and murderers, and for a time, they can seem invincible, but in the end, they always fall. Think of it, always.`,
  `My momma always said, “Life is like a box of chocolates, you never know what you’re gonna get.`
];

const REVIEW_TEXTS = [
  `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
  `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
  `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
  `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
  `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`

];

export {filmsCount, TITLES, POSTERS, GENRES, DIRECTORS, ACTORS, DESCRIPTION_TEXTS, REVIEW_TEXTS};
