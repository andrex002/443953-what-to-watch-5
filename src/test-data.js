export const film = {
  "id": 1,
  "title": "The Grand Budapest Hotel",
  "image": "img/the-grand-budapest-hotel-poster.jpg",
  "previewImage": "img/the-grand-budapest-hotel.jpg",
  "bgImage": "img/the-grand-budapest-hotel-bg.jpg",
  "bgColor": "#ffffff",
  "srcVideo": "https://some-link",
  "video": "https://some-link",
  "previewVideo": "https://some-link",
  "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
  "rating": {
    "score": 8.9,
    "count": 240
  },
  "director": "Wes Andreson",
  "actors": ["Bill Murray", "Edward Norton", "Jude Law", "Willem Dafoe", "Saoirse Ronan"],
  "duration": 99,
  "genre": "Comedy",
  "year": 2014,
  "isFavorite": false
};

export const films = [
  {
    "id": 1,
    "title": "The Grand Budapest Hotel",
    "image": "img/the-grand-budapest-hotel-poster.jpg",
    "previewImage": "img/the-grand-budapest-hotel.jpg",
    "bgImage": "img/the-grand-budapest-hotel-bg.jpg",
    "bgColor": "#ffffff",
    "video": "https://some-link",
    "previewVideo": "https://some-link",
    "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
    "rating": {
      "score": 8.9,
      "count": 240
    },
    "director": "Wes Andreson",
    "actors": ["Bill Murray", "Edward Norton", "Jude Law", "Willem Dafoe", "Saoirse Ronan"],
    "duration": 99,
    "genre": "Comedy",
    "year": 2014,
    "isFavorite": false
  },
  {
    "id": 2,
    "title": "The Grand Budapest Hotel",
    "image": "img/the-grand-budapest-hotel-poster.jpg",
    "previewImage": "img/the-grand-budapest-hotel.jpg",
    "bgImage": "img/the-grand-budapest-hotel-bg.jpg",
    "bgColor": "#ffffff",
    "video": "https://some-link",
    "previewVideo": "https://some-link",
    "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
    "rating": {
      "score": 8.9,
      "count": 240
    },
    "director": "Wes Andreson",
    "actors": ["Bill Murray", "Edward Norton", "Jude Law", "Willem Dafoe", "Saoirse Ronan"],
    "duration": 99,
    "genre": "Comedy",
    "year": 2014,
    "isFavorite": false
  }
];

export const comment = {
  "id": 1,
  "userId": 4,
  "userName": "Kate Muir",
  "rating": 8.9,
  "textComment": "Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.",
  "date": "2019-05-08T14:13:56.569Z"
};

export const comments = [
  {
    "id": 1,
    "userId": 4,
    "userName": "Kate Muir",
    "rating": 8.9,
    "textComment": "Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.",
    "date": "2019-05-08T14:13:56.569Z"
  },
  {
    "id": 2,
    "userId": 4,
    "userName": "Kate Muir",
    "rating": 8.9,
    "textComment": "Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.",
    "date": "2019-05-08T14:13:56.569Z"
  },
];

export const defaultState = {
  DATA: {
    allFilms: films,
    filteredFilms: films,
    promoFilm: film,
    currentFilm: film,
    favoriteFilms: films,
    comments: comments,
    numberFilmsShown: 8,
    isCommentSending: false,
    isCommentSendError: false
  },
  FILTER: {
    activeGenre: `All genres`,
  },
  USER: {
    authorizationStatus: `NO_AUTH`,
    name: `userName`,
    avatar: `avatar.jpg`
  }
};

export const noop = () => {};