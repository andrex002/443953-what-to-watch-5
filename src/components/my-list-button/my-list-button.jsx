import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {setIsFilmByIdLoading} from "../../store/action";
import {sendFavoriteStatus} from "../../store/api-action";
import {AuthorizationStatus} from "../../const";
import {Link} from "react-router-dom";

const MyListButton = (props) => {
  const {id, isFavorite, onMyListClick, authorizationStatus} = props;

  return (
    <Link className="btn btn--list movie-card__button" type="button"
      to={authorizationStatus === AuthorizationStatus.NO_AUTH ? `/login` : ``}
      onClick={(evt) => {
        if (authorizationStatus === AuthorizationStatus.AUTH) {
          evt.preventDefault();
          onMyListClick(id, isFavorite);
        }
      }}
    >
      {
        isFavorite
          ?
          <svg viewBox="0 0 19 20" width="19" height="20">
            <use xlinkHref="#in-list"></use>
          </svg>
          :
          <svg viewBox="0 0 19 20" width="19" height="20">
            <use xlinkHref="#add"></use>
          </svg>
      }
      <span>My list</span>
    </Link>
  );
};

MyListButton.propTypes = {
  id: PropTypes.number.isRequired,
  isFavorite: PropTypes.bool.isRequired,
  onMyListClick: PropTypes.func.isRequired,
  authorizationStatus: PropTypes.string.isRequired,
};

const mapStateToProps = ({USER}) => ({
  authorizationStatus: USER.authorizationStatus
});

const mapDispatchToProps = (dispatch) => ({
  onMyListClick(filmId, isFavorite) {
    dispatch(sendFavoriteStatus(filmId, isFavorite));
  }
});

export {MyListButton};
export default connect(mapStateToProps, mapDispatchToProps)(MyListButton);
