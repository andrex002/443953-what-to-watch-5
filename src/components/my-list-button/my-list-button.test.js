import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import {MyListButton} from "./my-list-button";
import {noop} from "../../test-data";
import {AuthorizationStatus} from "../../const";

describe(`Should MyListButton render correctly`, () => {
  it(`If film is favorite`, () => {
    const tree = renderer.create(
      <BrowserRouter>
        <MyListButton
          id={1}
          isFavorite={true}
          onMyListClick={noop}
          authorizationStatus={AuthorizationStatus.AUTH}
        />
      </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`If film is not favorite`, () => {
    const tree = renderer.create(
      <BrowserRouter>
        <MyListButton
          id={1}
          isFavorite={false}
          onMyListClick={noop}
          authorizationStatus={AuthorizationStatus.AUTH}
        />
      </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
