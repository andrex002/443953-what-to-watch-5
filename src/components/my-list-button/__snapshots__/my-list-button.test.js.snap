// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MyListButton render correctly If film is favorite 1`] = `
<a
  className="btn btn--list movie-card__button"
  href="/"
  onClick={[Function]}
  type="button"
>
  <svg
    height="20"
    viewBox="0 0 19 20"
    width="19"
  >
    <use
      xlinkHref="#in-list"
    />
  </svg>
  <span>
    My list
  </span>
</a>
`;

exports[`Should MyListButton render correctly If film is not favorite 1`] = `
<a
  className="btn btn--list movie-card__button"
  href="/"
  onClick={[Function]}
  type="button"
>
  <svg
    height="20"
    viewBox="0 0 19 20"
    width="19"
  >
    <use
      xlinkHref="#add"
    />
  </svg>
  <span>
    My list
  </span>
</a>
`;
